src/components/views/messages/DateSeparator.tsx

```diff
@@ -132,23 +132,67 @@ export default class DateSeparator extends React.Component<IProps, IState> {
                     `found ${eventId} (${originServerTs}) for timestamp=${unixTimestamp} (looking forward)`,
             );
 
-            dis.dispatch<ViewRoomPayload>({
-                action: Action.ViewRoom,
-                event_id: eventId,
-                highlighted: true,
-                room_id: roomId,
-                metricsTrigger: undefined, // room doesn't change
-            });
-        } catch (e) {
-            const code = e.errcode || e.statusCode;
-            // only show the dialog if failing for something other than a network error
-            // (e.g. no errcode or statusCode) as in that case the redactions end up in the
-            // detached queue and we show the room status bar to allow retry
-            if (typeof code !== "undefined") {
-                // display error message stating you couldn't delete this.
+            // Only try to navigate to the room if the user is still viewing the same
+            // room. We don't want to jump someone back to a room after a slow request
+            // if they've already navigated away to another room.
+            const roomIdBeforeNavigation = this.props.roomId;
```

Comment:
The room props are set when the timeline is created for a room so this doesn't change when the room changes.

We need another way to get the current/active room(s).

We could listen for any `Action.ViewRoom` in the interim but this isn't very nice and not compatible if we ever support viewing multiple rooms.

Reply:
It looks like we might have something for this:

`this.context.roomViewStore.getRoomId()`

`SdkContextClass.instance.roomViewStore.getRoomId()`